archipelago/Logic.py

```diff
@@ -1662,10 +1730,14 @@ def IsLevelEnterable(self, level):
 
     def WinConditionMet(self):
         """Check if the current game state has met the win condition."""
+        condition = self.settings.win_condition_item
+        # When using win condition-based ship spawning, always require K. Rool defeat in addition to win condition items
+        requires_krool_defeat = self.settings.krool_ship_spawn_method == 1
```

Comment:
Every instance of this guy is doing some weird python return-if statement which I kinda hate to read. I think it's simplest to just bake in the `Events.KRoolDefeated` into this variable and propagate it that way. Something like:
`krool_complete = self.settings.krool_ship_spawn_method != [that enum I mentioned earlier] or Events.KRoolDefeated in self.Events`